src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -92,4 +101,39 @@ class WorkflowManagerActor extends Actor with CromwellActor {
   private def idByWorkflow(workflow: WorkflowActorRef): Option[WorkflowId] = {
     workflowStore.toMap collectFirst {case (k, v) if v == workflow => k}
   }
+
+  private def restartIncompleteWorkflows(): Unit = {
+    // If the clob inputs for this workflow can be converted to JSON, return the JSON
+    // version of those inputs in a Some().  Otherwise return None.
+    def clobToJsonInputs(result: QueryWorkflowExecutionResult): Option[binding.WorkflowRawInputs] = {
+      result.wdlRawInputs.parseJson match {
+        case JsObject(rawInputs) => Option(rawInputs)
+        case x =>
+          log.error(s"Error restarting workflow ${result.workflowId}: expected JSON inputs, got '$x'.")
+          None
+      }
+    }
+    val RestartableStates = Some(Seq(WorkflowSubmitted, WorkflowRunning))
+    // Attempt to restart all the workflows in restartable states whose clob raw inputs
+    // can successfully be converted to JSON.
+    val restartableWorkflows = for {
+      workflow <- dataAccess.query(states = RestartableStates)
+      jsonInputs = clobToJsonInputs(workflow)
+      if jsonInputs.isDefined
+    } yield (workflow, jsonInputs.get)
+
+    val num = restartableWorkflows.length
+    val displayNum = if (num == 0) "no" else num
+    val plural = if (num == 1) "" else "s"
+    log.info(s"Found $displayNum workflow$plural to restart.")
+
+    if (num > 0) {
+      val ids = restartableWorkflows.map { _._1.workflowId.toString }.sorted
+      log.info(s"Starting workflow ID$plural: " + ids.mkString(", "))
+    }
```

Comment:
This was a somewhat arbitrary decision on my part.  If it's better (for some definition of better) to have N lines of logging of workflows being started instead of just 1, that can certainly be accomplished as you suggest.  The current code is consistent with the convention established by the call-starting logging, but that too could be changed.